src/cpp/qos/QosPolicies.cpp

```diff
@@ -39,6 +39,56 @@ uint32_t QosPolicy::get_cdr_serialized_size(
     return 2 + 2 + 4 + data_size;
 }
 
+bool QosPolicy::serialize_generic_data(
+        rtps::CDRMessage_t* msg,
+        uint16_t pid,
+        const std::vector<rtps::octet>& data)
+{
+    bool valid = CDRMessage::addUInt16(msg, pid);
+    uint16_t siz = static_cast<uint16_t>(data.size());
+    siz = (siz + 3) & ~3;
+    valid &= CDRMessage::addUInt16(msg, static_cast<uint16_t>(4 + siz));
+    valid &= CDRMessage::addUInt32(msg, siz);
+    valid &= CDRMessage::addOctetVector(msg, &data, true);
```

Comment:
`addOctetVector` is already inserting the size of the vector. Don't add it here too.

Reply:
👍